### DISCUSSION TITLE
The Game Gear Peel Out (aka The Strike Dash) + The Sonic Robo Blast 2 Thok!

### DISCUSSION THREAD

**Discussion by July2612** [TrianglyRU/OrbinautFramework#32] [Show and tell]
Hey there! After coding in the CD Spin Dash, I wanted to look for more things to add to my private fork of Orbinaut. Since I'm a fan of the Master System / Game Gear games, I really wanted to add in the version of the Peel Out that you see in Sonic Chaos and Triple Trouble, (as well as planned for Blast, but seemingly cut) also known as the Strike Dash! 

https://github.com/user-attachments/assets/2a0eec3b-829e-40bf-b826-b0eea44fb2ea

It's a version of the Peel Out you charge like a standard Spin Dash (although as far as I know, doing this in any of the Game Gear titles didn't affect the speed, so I decided to change that for this version for variable speeds), and once you release it, you get a short period of invulnerability!

In my private fork, there's not two macros for Dash Type, but _three_! One for the standard Sonic CD Peel Out (`DASH_CD`), and two for the slightly different versions of the Strike Dash from Chaos and Triple Trouble (`DASH_CHAOS`and `DASH_TTROUBLE`, respectively.)

![image](https://github.com/user-attachments/assets/0b8694d0-95bf-4651-ae74-02fada1187f5)

The minor difference is in how both games handle invulnerability. Sonic Chaos used standard I-Frames, aka the ones you'd have after getting hit, so you'd phase straight through badniks and hazards, and had no special effect. Sonic Triple Trouble, meanwhile, uses a small burst of invincibility similar to the monitor powerup, and Sonic would have stars similar to the power up as well as he charges and then uses the move, vanishing after the strike dash is done. Having both is cool, so I just made both! The version shown in the above video was the Triple Trouble version, down below is a video of the Sonic Chaos version. Notice how there's no stars, and Sonic goes through the Motobug without killing it.

https://github.com/user-attachments/assets/5ff4ea5a-3d4b-46ab-b382-e9fbbafe20db

But that's not all! After a stroke of genius, I decided to recreate the infamous jump ability from Sonic Robo Blast 2, the thok! (Apologies for the fact this is a low-framerate GIF and not a video like the others, filesize limits got in the way)

![ezgif-5-7812b6c189](https://github.com/user-attachments/assets/71b245a0-dec7-470c-8b8d-d002d469f7b8)

It's just a jump dash. Trust me, I tried to do the thing where there's a bit of an afterimage from where you thokked, but I couldn't make it work, I will try, though. It doesn't really work well with the insta-shield, since both activate at the same time, so I decided to just change the insta-shield global variable to a double jump type switcher, using macros `JUMP_NONE` for nothing, `JUMP_DSPIN `for the insta-shield, and `JUMP_THOK` for the thok.

![image](https://github.com/user-attachments/assets/9378b2e8-4183-4a4b-8c7d-0f998a43ae6a)

That's all for now! Soon, I'll make a tutorial on how to add all of the things from my private fork (CD Spin Dash, Strike Dash, Thok, and everything around them) into your Orbinaut project!! See ya later!

**Comment by RunninPigeon** [comment]
YESS PLEASEEEEEEEE

**Comment by Otukuoi** [comment]
For the last part, maybe you could just use the second action button to enable the Thok and the first to enable Insta-Shield and Drop-Dash? Seems like a pretty nifty way to get around that limitation.
